reactapp/src/Pages/components.js

```diff
@@ -0,0 +1,43 @@
+import React , { Component } from 'react';
+import {Link} from "react-router-dom";
+import '../styles/Header.css';
+import { connect } from 'react-redux';
+import '../styles/Main.css';
+
+class Item extends Component {
+    onClick () {
+        this.props.addToBucket(this.props.item.id);
+    }
+    render() {
+        return (
+            <div className={'Item'}>
+                <Link to={'/item/' + this.props.item.id}>
+                    <img src={'http://' + this.props.api.host + ':' + this.props.api.port + '/api/resources/' + this.props.item.id}/>
```

Comment:
You can make func "buildImageUrl()" for example and place it to utils. So you can use that like:
```
<img src={buildImageUri(id)} />
```
...